node/chainSimulator/components/testOnlyProcessingNode.go

```diff
@@ -382,26 +385,40 @@ func (node *testOnlyProcessingNode) collectClosableComponents(apiInterface APICo
 	}
 }
 
-// SetState will set the provided state for the given address
-func (node *testOnlyProcessingNode) SetState(address []byte, keyValueMap map[string]string) error {
+// SetKeyValueForAddress will set the provided state for the given address
+func (node *testOnlyProcessingNode) SetKeyValueForAddress(address []byte, keyValueMap map[string]string) error {
+	userAccount, err := node.getUserAccount(address)
+	if err != nil {
+		return err
+	}
+
+	err = setKeyValueMap(userAccount, keyValueMap)
+	if err != nil {
+		return err
+	}
+
 	accountsAdapter := node.StateComponentsHolder.AccountsAdapter()
-	account, err := accountsAdapter.LoadAccount(address)
+	err = accountsAdapter.SaveAccount(userAccount)
 	if err != nil {
 		return err
 	}
 
-	userAccount, ok := account.(state.UserAccountHandler)
-	if !ok {
-		return errors.New("cannot cast AccountHandler to UserAccountHandler")
+	_, err = accountsAdapter.Commit()
+	if err != nil {
```

Comment:
yes, as @sstanculeanu suggested, we need to return the err here and remove this if case

Reply:
done